yarn-project/p2p/src/mem_pools/instrumentation.ts

```diff
@@ -35,6 +44,26 @@ export class PoolInstrumentation<PoolObject extends Gossipable> {
         ],
       },
     });
+
+    this.dbMetrics = new LmdbMetrics(
+      meter,
+      {
+        name: Metrics.MEMPOOL_DB_MAP_SIZE,
+        description: 'Database map size for the Tx mempool',
+      },
+      {
+        name: Metrics.MEMPOOL_DB_USED_SIZE,
+        description: 'Database used size for the Tx mempool',
+      },
+      {
+        name: Metrics.MEMPOOL_DB_NUM_ITEMS,
+        description: 'Num items in database for the Tx mempool',
+      },
+    );
+  }
+
+  public recordDBMetrics(dbMapSize: number, dbNumItems: number, dbUsedSize: number) {
```

Comment:
It may be safer to accept a `{ mappingSize: number; actualSize: number; numItems: number }` object here, to ensure we don't mix up the fields. But it's just a nit.

Reply:
Agreed, will change